### PR TITLE
Adding Course pre and post install hooks

### PR DESCRIPTION
**Pull request by ejether**
refector of hooks, adding concept of course hooks, refactoring tests related to hooks, reorganizing some imports

### CODE REVIEW

**Comment by ejether** [comment]
Closes #130

File: reckoner/chart.py [hunk 1]
```diff
@@ -87,6 +89,26 @@ def __init__(self, chart, helm):
...
+            'Release {} pre install'.format(self.name),
+            self.config.course_base_directory
+        )
+
+        self._post_install_hook = Hook(
+            self.hooks.get(
+                'post_install',
+                []
+            ),
+            'Releae {} post install'.format(self.name),
```
Comment (bambash) [inline reckoner/chart.py]: typo on release

**Comment by ejether** [comment]
I admit, it's a little selfish as it enables my `fusion-cli` project but  hey, we all win. 
Here's hoping for a clean rebase :)

**Comment by ejether** [comment]
@sudermanjr  I'd like it if you could have a gander as well.

**Pull request merged**